### PR TITLE
[hotfix][docs] Update release notes for 1.13 with state backend migra…

### PR DESCRIPTION
**Pull request by sjwiesman**
…tion details

### CODE REVIEW

File: docs/content.zh/release-notes/flink-1.13.md [hunk 1]
```diff
@@ -38,6 +38,20 @@ by default before) and there is no option to configure a synchronous snapshot an
 The constructors of `FsStateBackend` and `MemoryStateBackend` that take a flag for sync/async
 snapshots are kept for API compatibility, but the flags are ignored now.
 
+##### [FLINK-19463](https://issues.apache.org/jira/browse/FLINK-19463)
+
+Flink has always separated local state storage from fault tolerance.
+Keyed state is maintained locally in state backends, either on the JVM heap or embedded RocksDB instances.
```
Comment (alpinegizmo) [inline docs/content.zh/release-notes/flink-1.13.md:44]:
```suggestion
Keyed state is maintained locally in state backends, either on the JVM heap or in embedded RocksDB instances.
```

File: docs/content.zh/release-notes/flink-1.13.md [hunk 2]
```diff
@@ -38,6 +38,20 @@ by default before) and there is no option to configure a synchronous snapshot an
...
+Fault tolerance comes from checkpoints and savepoints - periodic snapshots of the Jobs internal state to some durable file system - such as Amazon S3 or HDFS.
```
Comment (alpinegizmo) [inline docs/content.zh/release-notes/flink-1.13.md:45]:
```suggestion
Fault tolerance comes from checkpoints and savepoints - periodic snapshots of a job's internal state to some durable file system - such as Amazon S3 or HDFS. 
```

File: docs/content/release-notes/flink-1.13.md [hunk 3]
```diff
@@ -38,6 +38,22 @@ by default before) and there is no option to configure a synchronous snapshot an
 The constructors of `FsStateBackend` and `MemoryStateBackend` that take a flag for sync/async
 snapshots are kept for API compatibility, but the flags are ignored now.
 
+#### Disentangle StateBackends from Checkpointing
+
+##### [FLINK-19463](https://issues.apache.org/jira/browse/FLINK-19463)
+
+Flink has always separated local state storage from fault tolerance.
+Keyed state is maintained locally in state backends, either on the JVM heap or embedded RocksDB instances.
```
Comment (alpinegizmo) [inline docs/content/release-notes/flink-1.13.md:46]:
```suggestion
Keyed state is maintained locally in state backends, either on the JVM heap or in embedded RocksDB instances.
```

File: docs/content/release-notes/flink-1.13.md [hunk 4]
```diff
@@ -38,6 +38,22 @@ by default before) and there is no option to configure a synchronous snapshot an
...
+Fault tolerance comes from checkpoints and savepoints - periodic snapshots of the Jobs internal state to some durable file system - such as Amazon S3 or HDFS.
```
Comment (alpinegizmo) [inline docs/content/release-notes/flink-1.13.md:47]:
```suggestion
Fault tolerance comes from checkpoints and savepoints - periodic snapshots of a job's internal state to some durable file system - such as Amazon S3 or HDFS. 
```

**Review by alpinegizmo — CHANGES_REQUESTED** [review]
I think it would be good to also throw in a link to https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/ops/state/state_backends/#migrating-from-legacy-backends

**Comment by sjwiesman** [comment]
Changes applied

File: docs/content.zh/release-notes/flink-1.13.md [hunk 5]
```diff
@@ -38,6 +38,20 @@ by default before) and there is no option to configure a synchronous snapshot an
...
+Keyed state is maintained locally in state backends, either on the JVM heap or in embedded RocksDB instances.
+Fault tolerance comes from checkpoints and savepoints - periodic snapshots of a job's internal state to some durable file system - such as Amazon S3 or HDFS. 
+
+Historically, Flink's `StateBackend` interface intermixed these concepts in a way that confused many users. 
+In 1.13, checkpointing configurations have been extracted into their own interface, `CheckpointStorage`. 
+
+This change does not affect the runtime behavior and simply provides a better mental model to users. 
+Pipelines can be updated to use the new the new abstractions without losing state, consistency, or change in semantics. 
+
+Please follow the [migraiton guide](https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/ops/state/state_backends/#migrating-from-legacy-backends) or JavaDoc on the deprecated state backend classes - `MemoryStateBackend`, `FsStateBackend` and `RocksDBStateBackend` for migration details.
```
Comment (alpinegizmo) [inline docs/content.zh/release-notes/flink-1.13.md:53]:
```suggestion
Please follow the [migration guide](https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/ops/state/state_backends/#migrating-from-legacy-backends) or the JavaDoc on the deprecated state backend classes - `MemoryStateBackend`, `FsStateBackend` and `RocksDBStateBackend` for migration details. 
```

File: docs/content/release-notes/flink-1.13.md [hunk 6]
```diff
@@ -38,6 +38,22 @@ by default before) and there is no option to configure a synchronous snapshot an
...
+Keyed state is maintained locally in state backends, either on the JVM heap or in embedded RocksDB instances.
+Fault tolerance comes from checkpoints and savepoints - periodic snapshots of a job's internal state to some durable file system - such as Amazon S3 or HDFS. 
+
+Historically, Flink's `StateBackend` interface intermixed these concepts in a way that confused many users. 
+In 1.13, checkpointing configurations have been extracted into their own interface, `CheckpointStorage`. 
+
+This change does not affect the runtime behavior and simply provides a better mental model to users. 
+Pipelines can be updated to use the new the new abstractions without losing state, consistency, or change in semantics. 
+
+Please follow the [migraiton guide](https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/ops/state/state_backends/#migrating-from-legacy-backends) or JavaDoc on the deprecated state backend classes - `MemoryStateBackend`, `FsStateBackend` and `RocksDBStateBackend` for migration details.
```
Comment (alpinegizmo) [inline docs/content/release-notes/flink-1.13.md:55]:
```suggestion
Please follow the [migration guide](https://ci.apache.org/projects/flink/flink-docs-release-1.13/docs/ops/state/state_backends/#migrating-from-legacy-backends) or the JavaDoc on the deprecated state backend classes - `MemoryStateBackend`, `FsStateBackend` and `RocksDBStateBackend` for migration details. 
```

**Review by alpinegizmo — APPROVED** [review]
Please fix the misspelling (see the suggestions) before merging.

**Comment by sjwiesman** [comment]
merged into master and release-1.13

**Pull request closed without merging**